==11345== Memcheck, a memory error detector
==11345== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==11345== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==11345== Command: /home/wasmer/repositories/wasmer/wasmer/target/debug/examples/cross-compilation
==11345== 
--11345-- WARNING: unhandled amd64-linux syscall: 332
--11345-- You may be able to write your own handler.
--11345-- Read the file README_MISSING_SYSCALL_OR_IOCTL.
--11345-- Nevertheless we consider this a bug.  Please report
--11345-- it at http://valgrind.org/support/bug_reports.html.
==11345== 
==11345== HEAP SUMMARY:
==11345==     in use at exit: 29,448 bytes in 87 blocks
==11345==   total heap usage: 838 allocs, 751 frees, 123,338 bytes allocated
==11345== 
==11345== 32 bytes in 1 blocks are still reachable in loss record 1 of 27
==11345==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==11345==    by 0x4E3F165: dlsym (dlsym.c:70)
==11345==    by 0x1D66473: fetch (weak.rs:61)
==11345==    by 0x1D66473: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==11345==    by 0x1D66473: min_stack_size (thread.rs:449)
==11345==    by 0x1D66473: std::sys::unix::thread::Thread::new (thread.rs:49)
==11345==    by 0xEA05CB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345== 
==11345== 64 bytes in 4 blocks are still reachable in loss record 2 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE81536: new<crossbeam_deque::deque::Buffer<rayon_core::job::JobRef>> (boxed.rs:177)
==11345==    by 0xE81536: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==11345==    by 0xE9CD53: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==11345==    by 0xE9CEA3: crossbeam_epoch::atomic::Atomic<T>::init (atomic.rs:309)
==11345==    by 0xE9CE73: crossbeam_epoch::atomic::Atomic<T>::new (atomic.rs:294)
==11345==    by 0xE7FC36: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:253)
==11345==    by 0xE728FE: rayon_core::registry::Registry::new::{{closure}} (registry.rs:225)
==11345==    by 0xEAB64C: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE7A9CD: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345== 
==11345== 64 bytes in 4 blocks are still reachable in loss record 3 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D663CC: alloc (alloc.rs:80)
==11345==    by 0x1D663CC: alloc (alloc.rs:174)
==11345==    by 0x1D663CC: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D663CC: std::sys::unix::thread::Thread::new (thread.rs:44)
==11345==    by 0xEA05CB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345==    by 0xE94ED0: std::sync::once::Once::call_once (once.rs:264)
==11345==    by 0xE718B6: rayon_core::registry::set_global_registry (registry.rs:194)
==11345== 
==11345== 128 bytes in 4 blocks are still reachable in loss record 4 of 27
==11345==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==11345==    by 0x1D329FE: register_dtor (fast_thread_local.rs:33)
==11345==    by 0x1D329FE: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==11345==    by 0x1D329FE: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==11345==    by 0x1D551F9: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==11345==    by 0x1D551F9: __getit (local.rs:175)
==11345==    by 0x1D551F9: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==11345==    by 0x1D551F9: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==11345==    by 0x1D551F9: std::sys_common::thread_info::set (thread_info.rs:40)
==11345==    by 0xEA1A61: std::thread::Builder::spawn_unchecked::{{closure}} (mod.rs:473)
==11345==    by 0xE9607E: core::ops::function::FnOnce::call_once{{vtable-shim}} (function.rs:233)
==11345==    by 0x1D667E9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==11345==    by 0x1D667E9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==11345==    by 0x1D667E9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==11345==    by 0x52516DA: start_thread (pthread_create.c:463)
==11345==    by 0x57A271E: clone (clone.S:95)
==11345== 
==11345== 128 bytes in 4 blocks are still reachable in loss record 5 of 27
==11345==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==11345==    by 0xED98D1: std::thread::local::fast::Key<T>::try_register_dtor (local.rs:438)
==11345==    by 0xED97D2: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==11345==    by 0xED9976: std::thread::local::fast::Key<T>::get (local.rs:410)
==11345==    by 0xED96CC: crossbeam_epoch::default::HANDLE::__getit (local.rs:175)
==11345==    by 0xE84944: std::thread::local::LocalKey<T>::try_with (local.rs:262)
==11345==    by 0xE88DD2: crossbeam_epoch::default::with_handle (default.rs:43)
==11345==    by 0xE89036: crossbeam_epoch::default::is_pinned (default.rs:30)
==11345==    by 0xE7FDFC: crossbeam_deque::deque::Stealer<T>::steal (deque.rs:619)
==11345==    by 0xE74762: rayon_core::registry::WorkerThread::steal::{{closure}} (registry.rs:775)
==11345==    by 0xE6FBFC: core::iter::traits::iterator::Iterator::find_map::check::{{closure}} (iterator.rs:2258)
==11345== 
==11345== 160 bytes in 4 blocks are still reachable in loss record 6 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE9FDAF: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==11345==    by 0xEB34C0: rayon_core::latch::LockLatch::new (latch.rs:230)
==11345==    by 0xE736AC: rayon_core::registry::ThreadInfo::new (registry.rs:579)
==11345==    by 0xE95E71: core::ops::function::FnMut::call_mut (function.rs:154)
==11345==    by 0xEABE33: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE8EE53: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACB18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345==    by 0xEAACD9: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==11345== 
==11345== 160 bytes in 4 blocks are still reachable in loss record 7 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE9FDAF: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==11345==    by 0xEB34C0: rayon_core::latch::LockLatch::new (latch.rs:230)
==11345==    by 0xE736C4: rayon_core::registry::ThreadInfo::new (registry.rs:580)
==11345==    by 0xE95E71: core::ops::function::FnMut::call_mut (function.rs:154)
==11345==    by 0xEABE33: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE8EE53: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACB18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345==    by 0xEAACD9: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==11345== 
==11345== 160 bytes in 4 blocks are still reachable in loss record 8 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE9FDAF: std::sync::mutex::Mutex<T>::new (mutex.rs:223)
==11345==    by 0xEA3673: <std::sync::mutex::Mutex<T> as core::default::Default>::default (mutex.rs:446)
==11345==    by 0xE8E31E: <rayon_core::sleep::WorkerSleepState as core::default::Default>::default (mod.rs:55)
==11345==    by 0xE8B7CD: <crossbeam_utils::cache_padded::CachePadded<T> as core::default::Default>::default (cache_padded.rs:77)
==11345==    by 0xE8CBBA: rayon_core::sleep::Sleep::new::{{closure}} (mod.rs:67)
==11345==    by 0xEABB00: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE7ACA8: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACA59: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345== 
==11345== 160 bytes in 4 blocks are still reachable in loss record 9 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D33D9E: alloc (alloc.rs:80)
==11345==    by 0x1D33D9E: alloc (alloc.rs:174)
==11345==    by 0x1D33D9E: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D33D9E: new<()> (mutex.rs:223)
==11345==    by 0x1D33D9E: std::thread::Thread::new (mod.rs:1149)
==11345==    by 0xEA036D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345==    by 0xE94ED0: std::sync::once::Once::call_once (once.rs:264)
==11345==    by 0xE718B6: rayon_core::registry::set_global_registry (registry.rs:194)
==11345== 
==11345== 160 bytes in 4 blocks are still reachable in loss record 10 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE6A9B7: alloc::sync::Arc<T>::new (sync.rs:324)
==11345==    by 0xEA040B: std::thread::Builder::spawn_unchecked (mod.rs:465)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345== 
==11345== 192 bytes in 4 blocks are still reachable in loss record 11 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D50E96: alloc (alloc.rs:80)
==11345==    by 0x1D50E96: alloc (alloc.rs:174)
==11345==    by 0x1D50E96: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D50E96: std::sync::condvar::Condvar::new (condvar.rs:130)
==11345==    by 0xEB34DC: rayon_core::latch::LockLatch::new (latch.rs:231)
==11345==    by 0xE736AC: rayon_core::registry::ThreadInfo::new (registry.rs:579)
==11345==    by 0xE95E71: core::ops::function::FnMut::call_mut (function.rs:154)
==11345==    by 0xEABE33: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE8EE53: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACB18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345==    by 0xEAACD9: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==11345==    by 0xE92CD1: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==11345==    by 0xE933B4: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==11345==    by 0xE944E6: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==11345== 
==11345== 192 bytes in 4 blocks are still reachable in loss record 12 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D50E96: alloc (alloc.rs:80)
==11345==    by 0x1D50E96: alloc (alloc.rs:174)
==11345==    by 0x1D50E96: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D50E96: std::sync::condvar::Condvar::new (condvar.rs:130)
==11345==    by 0xEB34DC: rayon_core::latch::LockLatch::new (latch.rs:231)
==11345==    by 0xE736C4: rayon_core::registry::ThreadInfo::new (registry.rs:580)
==11345==    by 0xE95E71: core::ops::function::FnMut::call_mut (function.rs:154)
==11345==    by 0xEABE33: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE8EE53: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACB18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345==    by 0xEAACD9: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==11345==    by 0xE92CD1: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==11345==    by 0xE933B4: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==11345==    by 0xE944E6: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==11345== 
==11345== 192 bytes in 4 blocks are still reachable in loss record 13 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D50E96: alloc (alloc.rs:80)
==11345==    by 0x1D50E96: alloc (alloc.rs:174)
==11345==    by 0x1D50E96: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D50E96: std::sync::condvar::Condvar::new (condvar.rs:130)
==11345==    by 0xE8E33A: <rayon_core::sleep::WorkerSleepState as core::default::Default>::default (mod.rs:57)
==11345==    by 0xE8B7CD: <crossbeam_utils::cache_padded::CachePadded<T> as core::default::Default>::default (cache_padded.rs:77)
==11345==    by 0xE8CBBA: rayon_core::sleep::Sleep::new::{{closure}} (mod.rs:67)
==11345==    by 0xEABB00: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE7ACA8: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACA59: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345==    by 0xEAABAA: core::iter::traits::iterator::Iterator::for_each (iterator.rs:658)
==11345==    by 0xE928D7: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2140)
==11345==    by 0xE93501: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==11345==    by 0xE94545: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==11345== 
==11345== 192 bytes in 4 blocks are still reachable in loss record 14 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D33E02: alloc (alloc.rs:80)
==11345==    by 0x1D33E02: alloc (alloc.rs:174)
==11345==    by 0x1D33E02: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D33E02: new (condvar.rs:130)
==11345==    by 0x1D33E02: std::thread::Thread::new (mod.rs:1150)
==11345==    by 0xEA036D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345==    by 0xE94ED0: std::sync::once::Once::call_once (once.rs:264)
==11345==    by 0xE718B6: rayon_core::registry::set_global_registry (registry.rs:194)
==11345== 
==11345== 320 bytes in 4 blocks are still reachable in loss record 15 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D33E51: alloc (alloc.rs:80)
==11345==    by 0x1D33E51: alloc (alloc.rs:174)
==11345==    by 0x1D33E51: exchange_malloc (alloc.rs:277)
==11345==    by 0x1D33E51: new<std::thread::Inner> (sync.rs:324)
==11345==    by 0x1D33E51: std::thread::Thread::new (mod.rs:1145)
==11345==    by 0xEA036D: std::thread::Builder::spawn_unchecked (mod.rs:462)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345==    by 0xE94ED0: std::sync::once::Once::call_once (once.rs:264)
==11345==    by 0xE718B6: rayon_core::registry::set_global_registry (registry.rs:194)
==11345== 
==11345== 384 bytes in 1 blocks are still reachable in loss record 16 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xF0223B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xF02AE3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xEF95FA: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==11345==    by 0xEB6940: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==11345==    by 0xEB4B53: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==11345==    by 0xEB8412: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==11345==    by 0xE916F8: alloc::vec::Vec<T>::reserve (vec.rs:505)
==11345==    by 0xE92BE7: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2136)
==11345==    by 0xE933B4: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==11345==    by 0xE944E6: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter (vec.rs:1995)
==11345==    by 0xEAA968: core::iter::traits::iterator::Iterator::collect (iterator.rs:1671)
==11345== 
==11345== 416 bytes in 4 blocks are still reachable in loss record 17 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xEA0537: new<closure-0> (boxed.rs:177)
==11345==    by 0xEA0537: std::thread::Builder::spawn_unchecked (mod.rs:494)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345== 
==11345== 512 bytes in 1 blocks are still reachable in loss record 18 of 27
==11345==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D56909: aligned_malloc (alloc.rs:95)
==11345==    by 0x1D56909: alloc (alloc.rs:22)
==11345==    by 0x1D56909: __rdl_alloc (alloc.rs:304)
==11345==    by 0xF0223B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xF02AE3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xEF95FA: alloc::raw_vec::finish_grow (raw_vec.rs:497)
==11345==    by 0xEB65E0: alloc::raw_vec::RawVec<T,A>::grow_amortized (raw_vec.rs:429)
==11345==    by 0xEB4A53: alloc::raw_vec::RawVec<T,A>::try_reserve (raw_vec.rs:310)
==11345==    by 0xEB8552: alloc::raw_vec::RawVec<T,A>::reserve (raw_vec.rs:300)
==11345==    by 0xE916C8: alloc::vec::Vec<T>::reserve (vec.rs:505)
==11345==    by 0xE92817: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::spec_extend (vec.rs:2136)
==11345==    by 0xE93501: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (vec.rs:2120)
==11345== 
==11345== 640 bytes in 1 blocks are still reachable in loss record 19 of 27
==11345==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D56909: aligned_malloc (alloc.rs:95)
==11345==    by 0x1D56909: alloc (alloc.rs:22)
==11345==    by 0x1D56909: __rdl_alloc (alloc.rs:304)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE6A7E5: alloc::sync::Arc<T>::new (sync.rs:324)
==11345==    by 0xE72077: rayon_core::registry::Registry::new (registry.rs:234)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345== 
==11345== 640 bytes in 1 blocks are still reachable in loss record 20 of 27
==11345==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D56909: aligned_malloc (alloc.rs:95)
==11345==    by 0x1D56909: alloc (alloc.rs:22)
==11345==    by 0x1D56909: __rdl_alloc (alloc.rs:304)
==11345==    by 0xEDE13B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xEDE683: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xEDE09C: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xEDA315: alloc::sync::Arc<T>::new (sync.rs:324)
==11345==    by 0xEDA859: <crossbeam_epoch::collector::Collector as core::default::Default>::default (collector.rs:32)
==11345==    by 0xEDA886: crossbeam_epoch::collector::Collector::new (collector.rs:40)
==11345==    by 0xED8D79: __static_ref_initialize (default.rs:13)
==11345==    by 0xED8D79: core::ops::function::FnOnce::call_once (function.rs:233)
==11345==    by 0xED8BE5: lazy_static::lazy::Lazy<T>::get::{{closure}} (inline_lazy.rs:31)
==11345==    by 0xED9EA4: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345== 
==11345== 1,152 bytes in 4 blocks are possibly lost in loss record 21 of 27
==11345==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==11345==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==11345==    by 0x5252227: allocate_stack (allocatestack.c:627)
==11345==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==11345==    by 0x1D6651C: std::sys::unix::thread::Thread::new (thread.rs:66)
==11345==    by 0xEA05CB: std::thread::Builder::spawn_unchecked (mod.rs:492)
==11345==    by 0xEA284B: std::thread::Builder::spawn (mod.rs:386)
==11345==    by 0xE71338: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (registry.rs:100)
==11345==    by 0xE7247F: rayon_core::registry::Registry::new (registry.rs:256)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345== 
==11345== 1,520 bytes in 1 blocks are still reachable in loss record 22 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE81911: new<crossbeam_deque::deque::Block<rayon_core::job::JobRef>> (boxed.rs:177)
==11345==    by 0xE81911: <crossbeam_deque::deque::Injector<T> as core::default::Default>::default (deque.rs:1187)
==11345==    by 0xE8019D: crossbeam_deque::deque::Injector<T>::new (deque.rs:1213)
==11345==    by 0xE71E43: rayon_core::registry::Registry::new (registry.rs:238)
==11345==    by 0xE715AD: rayon_core::registry::global_registry::{{closure}} (registry.rs:168)
==11345==    by 0xE71A69: rayon_core::registry::set_global_registry::{{closure}} (registry.rs:195)
==11345==    by 0xE95133: std::sync::once::Once::call_once::{{closure}} (once.rs:264)
==11345==    by 0x1D51549: std::sync::once::Once::call_inner (once.rs:416)
==11345==    by 0xE94ED0: std::sync::once::Once::call_once (once.rs:264)
==11345== 
==11345== 1,536 bytes in 4 blocks are still reachable in loss record 23 of 27
==11345==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0x1D56909: aligned_malloc (alloc.rs:95)
==11345==    by 0x1D56909: alloc (alloc.rs:22)
==11345==    by 0x1D56909: __rdl_alloc (alloc.rs:304)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE6AB45: alloc::sync::Arc<T>::new (sync.rs:324)
==11345==    by 0xE7FCCD: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:250)
==11345==    by 0xE728FE: rayon_core::registry::Registry::new::{{closure}} (registry.rs:225)
==11345==    by 0xEAB64C: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE7A9CD: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345==    by 0xEACC03: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (mod.rs:873)
==11345== 
==11345== 2,008 bytes in 1 blocks are still reachable in loss record 24 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xEDE13B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xEDE683: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xEDE09C: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xED6A59: new<crossbeam_epoch::sync::queue::Node<crossbeam_epoch::internal::SealedBag>> (boxed.rs:177)
==11345==    by 0xED6A59: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==11345==    by 0xED6405: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==11345==    by 0xED6355: crossbeam_epoch::atomic::Owned<T>::new (atomic.rs:846)
==11345==    by 0xED6C5B: crossbeam_epoch::sync::queue::Queue<T>::new (queue.rs:50)
==11345==    by 0xED2053: crossbeam_epoch::internal::Global::new (internal.rs:254)
==11345==    by 0xEDA84B: <crossbeam_epoch::collector::Collector as core::default::Default>::default (collector.rs:32)
==11345==    by 0xEDA886: crossbeam_epoch::collector::Collector::new (collector.rs:40)
==11345==    by 0xED8D79: __static_ref_initialize (default.rs:13)
==11345==    by 0xED8D79: core::ops::function::FnOnce::call_once (function.rs:233)
==11345== 
==11345== 4,096 bytes in 4 blocks are still reachable in loss record 25 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xEB459D: alloc::raw_vec::RawVec<T,A>::allocate_in (raw_vec.rs:183)
==11345==    by 0xEB7B00: alloc::raw_vec::RawVec<T,A>::with_capacity_in (raw_vec.rs:159)
==11345==    by 0xEB3C7E: alloc::raw_vec::RawVec<T>::with_capacity (raw_vec.rs:90)
==11345==    by 0xE8F49E: alloc::vec::Vec<T>::with_capacity (vec.rs:363)
==11345==    by 0xE7E6E0: crossbeam_deque::deque::Buffer<T>::alloc (deque.rs:45)
==11345==    by 0xE7FBB9: crossbeam_deque::deque::Worker<T>::new_lifo (deque.rs:248)
==11345==    by 0xE728FE: rayon_core::registry::Registry::new::{{closure}} (registry.rs:225)
==11345==    by 0xEAB64C: core::iter::adapters::map_fold::{{closure}} (mod.rs:833)
==11345==    by 0xE7A9CD: core::iter::traits::iterator::Iterator::fold (iterator.rs:2022)
==11345== 
==11345== 6,080 bytes in 4 blocks are still reachable in loss record 26 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xE9B08B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xE9BEB3: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xE9AFEC: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xE81911: new<crossbeam_deque::deque::Block<rayon_core::job::JobRef>> (boxed.rs:177)
==11345==    by 0xE81911: <crossbeam_deque::deque::Injector<T> as core::default::Default>::default (deque.rs:1187)
==11345==    by 0xE8019D: crossbeam_deque::deque::Injector<T>::new (deque.rs:1213)
==11345==    by 0xE6C57C: rayon_core::job::JobFifo::new (job.rs:193)
==11345==    by 0xE7489A: rayon_core::registry::main_loop (registry.rs:802)
==11345==    by 0xE70E6F: rayon_core::registry::ThreadBuilder::run (registry.rs:55)
==11345==    by 0xE71520: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn::{{closure}} (registry.rs:100)
==11345==    by 0xEA9F4F: std::sys_common::backtrace::__rust_begin_short_backtrace (backtrace.rs:130)
==11345==    by 0xEA2750: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}} (mod.rs:475)
==11345== 
==11345== 8,160 bytes in 4 blocks are still reachable in loss record 27 of 27
==11345==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11345==    by 0xEDE13B: alloc::alloc::alloc (alloc.rs:80)
==11345==    by 0xEDE683: <alloc::alloc::Global as core::alloc::AllocRef>::alloc (alloc.rs:174)
==11345==    by 0xEDE09C: alloc::alloc::exchange_malloc (alloc.rs:277)
==11345==    by 0xED1A89: new<crossbeam_epoch::internal::Local> (boxed.rs:177)
==11345==    by 0xED1A89: <T as crossbeam_epoch::atomic::Pointable>::init (atomic.rs:188)
==11345==    by 0xED63A5: crossbeam_epoch::atomic::Owned<T>::init (atomic.rs:861)
==11345==    by 0xED6305: crossbeam_epoch::atomic::Owned<T>::new (atomic.rs:846)
==11345==    by 0xED28F2: crossbeam_epoch::internal::Local::register (internal.rs:392)
==11345==    by 0xEDA8AD: crossbeam_epoch::collector::Collector::register (collector.rs:45)
==11345==    by 0xED9693: crossbeam_epoch::default::HANDLE::__init (default.rs:18)
==11345==    by 0xED8D58: core::ops::function::FnOnce::call_once (function.rs:233)
==11345==    by 0xEDE82B: std::thread::local::lazy::LazyKeyInner<T>::initialize (local.rs:289)
==11345== 
==11345== LEAK SUMMARY:
==11345==    definitely lost: 0 bytes in 0 blocks
==11345==    indirectly lost: 0 bytes in 0 blocks
==11345==      possibly lost: 1,152 bytes in 4 blocks
==11345==    still reachable: 28,296 bytes in 83 blocks
==11345==         suppressed: 0 bytes in 0 blocks
==11345== 
==11345== For counts of detected and suppressed errors, rerun with: -v
==11345== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
